environment.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+from dotenv import dotenv_values
+
+
+def get_es_config(raise_on_missing=False):
+    env_config = {
+        **dotenv_values('.env'),
+        **os.environ,  # Override values loaded from file with those set in shell (if any)
+    }
+
+    es_config = {}
+
+    elastic_cloud_id = env_config.get('ELASTIC_CLOUD_ID', None)
+    elastic_url = env_config.get('ELASTIC_URL', None)
+
+    if elastic_cloud_id and elastic_url:
+        print(
+            '**Warning!!!**\n'
+            'Both ELASTIC_CLOUD_ID and ELASTIC_URL are set. Using the value from ELASTIC_CLOUD_ID.'
+        )
+        es_config['cloud_id'] = elastic_cloud_id
+    elif elastic_cloud_id:
+        es_config['cloud_id'] = elastic_cloud_id
+    elif elastic_url:
+        es_config['hosts'] = [elastic_url]
+    elif raise_on_missing:
+        raise RuntimeError('At least one of ELASTIC_URL or ELASTIC_CLOUD_ID must be set in environment variables.')
+    else:
+        print(
+            '**Warning!!!**\n'
+            'Neither ELASTIC_URL nor ELASTIC_CLOUD_ID is set. Will use the default value of "http://localhost:9200".'
+        )
+        es_config['hosts'] = ['http://localhost:9200']
+    
+    if 'ELASTIC_USER' not in env_config:
+        print(
+            '**Warning!!!**\n'
+            'ELASTIC_USER is not set. Will use the default value of "elastic".'
+        )
+    if 'ELASTIC_PASSWORD' not in env_config:
+        print(
+            '**Warning!!!**\n'
+            'ELASTIC_PASSWORD is not set. Will pass `None` to ElasticSearch client (which will probably fail).'
+        )
+
+    es_config['basic_auth'] = (env_config.get('ELASTIC_USER', 'elastic'), env_config.get('ELASTIC_PASSWORD', None))
+    
+    return es_config
+
+
+def get_env_vars(raise_on_missing=False):
+    # Entries in this list are required.
+    needed = ['OPENAI_KEY']
+    # Entries in this list are optional/less important.
+    wanted = ['GOOGLE_MAPS_KEY']
```

Comment:
this is fine for now, but eventually this needs to be more arbiturary: see how the top of `api.py` is implemented. The hope is to have many many providers, each of which may carry their set of configurations. Hence, it is probably eventually good to merge this all into `simon.providers`.

Reply:
Yeah it could also be possible to change this to functions that directly instantiate clients (e.g. return `ElasticSearch` instances directly rather than config) but don't want to overthink things too much too early